README.md

```diff
@@ -73,12 +159,221 @@ new RollbarConfig
+    }
+} 
+```
+
+All the scoped instances of the Notifier component are `IDisposable`, so use the `using(…){…}` block while dealing with short-lived instances or call the `Dispose()` method manually when controlling the instance lifetime manually.
 
-To send a caught exception to Rollbar, you must call Rollbar.Report(). You can set an item's level when you call this function. The level can be 'Debug', 'Info', 'Warning', 'Error' and 'Critical'.
+### RollbarLocator
+
+This is a helper singleton-like type used to gain access and instantiate the special singleton-like instance of the Notifier on the first access. Call its `RollbarLocator.RollbarInstance` property to get reference to the special Notifier singleton instance.
 
 ```csharp
-Rollbar.Report(exception, ErrorLevel.Critical);
+RollbarLocator.RollbarInstance
+    .Configure(new RollbarConfig("POST_SERVER_ACCESS_TOKEN") { Environment = "production" })
+    ;
+```
+
+### RollbarQueueController
+
+This is a singleton type used to gain access to the SDK-wide internal events at runtime. 
+
+To monitor any Notifier library internal event, regardless of the specific instance of the Notifier:
+
+1. Get a reference to the `RollbarQueueController.Instance` singleton.
+2. Subscribe to its `InternalEvent` event.
+3. Implement the event handler as you see fit.
+At runtime, all the Rollbar internal events generated while using any instance of the Notifier will be reported into the event handler. 
+
+To monitor internal events within any specific instance of the Notifier:
+1. Get a reference to a specific instance of the Notifier. 
+2. Subscribe to its `InternalEvent` event.
+3, Implement the event handler as you see fit.
+At runtime, all the Rollbar internal events generated while using this specific instance of the Notifier will be reported into the event handler. 
+
+An example demonstrating both levels of monitoring at the same time:
+
+```csharp
+static void Main(string[] args)
+{
+    RollbarQueueController.Instance.InternalEvent += OnRollbarInternalEvent;
+
+    const string postServerItemAccessToken = "POST_SERVER_ACCESS_TOKEN";
+
+    RollbarLocator.RollbarInstance
+        .Configure(new RollbarConfig(postServerItemAccessToken) { Environment = "proxyTest" })
+        .InternalEvent += OnRollbarInternalEvent
+        ;
+
+    RollbarLocator.RollbarInstance
+        .Info("Basic info log example.")
+        .Debug("First debug log.")
+        .Error(new NullReferenceException())
+        .Error(new Exception("trying out the TraceChain", new NullReferenceException()))
+        ;
+
+    System.Threading.Thread.Sleep(TimeSpan.FromSeconds(10));
+}
+
+private static void OnRollbarInternalEvent(object sender, RollbarEventArgs e)
+{
+    Console.WriteLine(e.TraceAsString());
+
+    RollbarApiErrorEventArgs apiErrorEvent = e as RollbarApiErrorEventArgs;
+    if (apiErrorEvent != null)
+    {
+        //TODO: handle/report Rollbar API communication error event...
```

Comment:
Who are these TODO notes for? They were in the docs given to me by @akornich but it isn't clear to me.

Reply:
that is for the SDK users to place whatever code they want to use to handle the SDK's internal errors/events. That is a boilerplate sample...

Reply:
okay got it, thank you!